src/cogent3/util/io.py

```diff
@@ -343,8 +344,81 @@ def remove_files(list_of_filepaths, error_on_missing=True):
         raise OSError("Some filepaths were not accessible: %s" % "\t".join(missing))
 
 
-def path_exists(path: os.PathLike) -> bool:
+def path_exists(path: PathType) -> bool:
     """whether path is a valid path and it exists"""
-    with contextlib.suppress(ValueError, TypeError):
-        return os_path.exists(str(path))
+    with contextlib.suppress(Exception):
+        return Path(path).exists()
     return False
+
+
+def iter_splitlines(
+    path: PathType, chunk_size: Optional[int] = 1_000_000
+) -> Generator[str, None, None]:
```

Comment:
Fine as it is, [docs](https://docs.python.org/3/library/typing.html#typing.Generator) list that when there is no `SendType` or `ReturnType`, it could also be simplified to either `Iterator[YieldType]` or `Iterable[YieldType]`